AlicerceTests/Sources/Network/URLSessionNetworkStackTestCase.swift

```diff
@@ -0,0 +1,192 @@
+            } catch Network.Error.badResponse {
+                // 🤠 well done sir
+            } catch {
+                XCTFail("🔥 received unexpected error 👉 \(error) 😱")
+            }
+        }
+    }
+
+    func testURLSession_WhenResponseDontHasASuccessfulStatusCode_ItShouldThrowStatusCodeError() {
+        let baseURL = URL(string: "http://")!
+
+        let statusCode = 500
+
+        let mockSession = MockURLSession()
+        mockSession.mockURLResponse = HTTPURLResponse(url: baseURL,
+                                                      statusCode: statusCode,
+                                                      httpVersion: nil,
+                                                      headerFields: nil)!
+
+        let network = Network.URLSessionNetworkStack(baseURL: baseURL, session: mockSession)
+
+        network.fetch(resource: resource) { (inner: () throws -> Data) -> Void in
+            do {
+                let _ = try inner()
+
+                XCTFail("🔥 should throw an error 🤔")
+            } catch let Network.Error.http(code: receiveStatusCode, description: _) {
+                XCTAssertEqual(receiveStatusCode.rawValue, statusCode, "✅ received same sent status code from network 🤠")
+            } catch {
+                XCTFail("🔥 received unexpected error 👉 \(error) 😱")
+            }
+        }
+    }
+
+    func testURLSession_WhenResponseHasNoData_ItShouldThrowANoDataError() {
+        let baseURL = URL(string: "http://")!
+
+        let statusCode = 200
+
+        let mockSession = MockURLSession()
+        mockSession.mockURLResponse = HTTPURLResponse(url: baseURL,
+                                                      statusCode: statusCode,
+                                                      httpVersion: nil,
+                                                      headerFields: nil)!
+        mockSession.mockDataTaskData = nil
+
+        let network = Network.URLSessionNetworkStack(baseURL: baseURL, session: mockSession)
+
+        network.fetch(resource: resource) { (inner: () throws -> Data) -> Void in
+            do {
+                let _ = try inner()
+
+                XCTFail("🔥 should throw an error 🤔")
+            } catch Network.Error.noData {
+                // 🤠 well done sir
+            } catch {
+                XCTFail("🔥 received unexpected error 👉 \(error) 😱")
+            }
+        }
+    }
+}
+
+
+// MARK: - Network Mocks
+
+final class MockURLSession : URLSession {
+
+    var mockDataTaskData: Data? = Data()
+    var mockDataTaskError: Error? = nil
+    var mockURLResponse: URLResponse = URLResponse()
+
+    override func dataTask(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask {
+
+        let dataTask = MockURLSessionDataTask()
+
+        dataTask.resumeInvokedClosure = { [weak self] in
+            guard let `self` = self else { fatalError("🔥: `self` must be defined!") }
```

Comment:
I could be wrong, but isn't the use of `'self' = self` going to be removed soon, as its a hack? Should we preempt this and use `strongSelf` instead?